### PR TITLE
UniConfig API callbacks, GlobalProvider removal

### PR DESCRIPTION
**Pull request by ChmaraX**
There are only some basic types on few API calls - further work required in upcoming PRs. 

Signed-off-by: Adam Chmara <adam.chmara1@gmail.com>

### CODE REVIEW

File: packages/frinx-dashboard/src/api/uniconfig/api-helpers.ts [hunk 1]
```diff
@@ -9,12 +10,19 @@ export async function apiFetch(path: string, options: RequestInit): Promise<unkn
     throw new Error(`apiFetch failed with http-code ${response.status}`);
   }
 
+  if (response.status === 201 || response.status === 204) {
+    return response;
```
Comment (Paulooze) [inline packages/frinx-dashboard/src/api/uniconfig/api-helpers.ts:14]: `return null` should be enough, we don't need to return the `response` object here.

File: packages/frinx-dashboard/src/api/uniconfig/api-helpers.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // TODO dynamic source of url
```
Comment (Paulooze) [inline packages/frinx-dashboard/src/api/uniconfig/api-helpers.ts:1]: This is outdated I guess, so remove it.

File: packages/frinx-dashboard/src/api/uniconfig/api-helpers.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
...
-const UNICONFIG_API_URL = 'localhost:8080/api';
+const UNICONFIG_API_URL = window.__CONFIG__.uniconfig_api_url;
+const AUTHORIZATION = 'Basic YWRtaW46YWRtaW4=';
```
Comment (Paulooze) [inline packages/frinx-dashboard/src/api/uniconfig/api-helpers.ts:3]: Maybe add a comment explaining why this is here. I have no idea if this is a temporary "hack", or a production solution.

File: packages/frinx-dashboard/src/uniconfig-app.tsx [hunk 4]
```diff
@@ -1,82 +1,153 @@
...
+  replaceConfigWithOperational,
+  getSnapshots,
+  deleteSnapshot,
+  replaceConfigWithSnapshot,
+  createSnapshot,
+};
 
 const UniconfigApp: FC = () => {
   const history = useHistory();
+  const query = new URLSearchParams(useLocation().search);
```
Comment (Paulooze) [inline packages/frinx-dashboard/src/uniconfig-app.tsx:57]:
```typescript
const location = useLocation();
const query = new URLSearchParams(location.search);
```

File: yarn.lock [hunk 5]
```diff
@@ -12831,12 +12831,20 @@ query-string@^4.1.0:
     strict-uri-encode "^1.0.0"
 
 query-string@^6.13.7:
+<<<<<<< Updated upstream
```
Comment (Paulooze) [inline yarn.lock:12834]: Please run `yarn install` in the root and commit `yarn.lock` file without conflicts.

**Review by Paulooze — CHANGES_REQUESTED** [review]

File: packages/frinx-dashboard/src/api/uniconfig/api-helpers.ts (hunk 1 above)
Comment (ChmaraX) [reply packages/frinx-dashboard/src/api/uniconfig/api-helpers.ts:14]: Response statusText and code is used in some cases as alerts so let's leave it here for now.

Comment (Paulooze) [reply packages/frinx-dashboard/src/api/uniconfig/api-helpers.ts:13]: Where? How? This is a response in a case when the status code is 201 or 204, there is no need for status text and/or code? Just show me where/how it's used and I am sure we can find something better.

**Review by Paulooze — APPROVED** [review]

**Pull request merged**